packages/frameworks/index.d.ts

```diff
@@ -3,7 +3,9 @@ export interface FrameworkDetectionItem {
   matchContent?: string;
 }
 
-type Setting = { value: string } | { placeholder: string };
+type Setting =
```

Comment:
This seems hacky to me. The proper TypeScript way to handle the disambiguation is to use a type guard function to narrow the type:

```js
interface SettingPlaceholder { placeholder: string; }
interface SettingValue { value: string; }
type Setting = SettingValue | SettingPlaceholder;

function isSettingPlaceholder(v: any): v is SettingPlaceholder {
  return v && typeof v.placeholder === 'string';
}

//

if (isPlaceholderSetting(setting)) {
  // `setting.placeholder` can safely be used
} else {
  // `setting.value` can safely be used
}
```

Reply:
100% agree! I'll update in a follow-up PR.